manifests/host.pp

```diff
@@ -1,7 +1,7 @@
 # == Define: dhcp::host
 #
 define dhcp::host (
-  Optional[Stdlib::IP::Address] $ip     = undef,
+  Optional[Stdlib::Host] $ip            = undef,
```

Comment:
The naming of the parameter must also be changed in my opinion. From a user point of view still having to specify ip parameter when I can also sent FQDN towards it is a bit confusing. This does however causes some backwards compatability issue's. My suggestion would be to implement a new parameter called $host which is of type stdlib::host make a check, which checks if either $ip or $host is set. If $ip is set raise a deprication warning that $ip will be removed in the next version.

What do others think of this?

Reply:
@zyronix agreed, don't forget to update the documentation!

Reply:
I agree that as a user this would confuse me and a rename is in order.

Reply:
I apologize for taking so long to respond.  I agree that the `$ip` parameter should be changed to something like `$host`.  I originally left it as-is since it would probably break a lot of users `dhcp::host:` configuration.  I can provide some unit tests once I get those setup.  Due to time constraints, it might take me until next week.